app/client/components/set_selector.es6

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import Select from 'react-select';
+import { Async } from 'react-select';
+
+class SetSelector extends React.Component {
+
+  state = {
+    selectedOption: null
+  }
+
+  constructor(props) {
+    super()
+    this.onOpenOptions = props.onOpenOptions
+    this.onCloseOptions = props.onCloseOptions
+    this.onChange = props.onChange
+    this.onChangeOption = this.onChangeOption.bind(this)
+    this.getOptions = this.getOptions.bind(this)
+  }
+
+  convertOpts(json) {
+    const selectedOption = this.state.selectedOption
+    if (!json.data) {
+      return [selectedOption]
+    }
+    return json.data.map((o)=>{ 
+      const label = o.attributes.name
+      const value = o.id
+      return {
+        label, value
+      }
+    })
+  }
+
+  onChangeOption(selectedOption) {
+    this.setState({ selectedOption })
+    this.onChange(selectedOption)
+  }
+
+  getOptions(value) {
+    return fetch(`/sets_service/sets?filter[search_by_name]=${value}`)
```

Comment:
I think we should also filter by locked sets, as those are showing up in the results